app/models/review.rb

```diff
@@ -0,0 +1,10 @@
+class Review < ActiveRecord::Base
+  belongs_to :bar
+  belongs_to :user
+
+  validates :title, presence: true
+  validates :body, presence: true
+  validates :bar_id, presence: true
+  validates :user_id, presence: true
```

Comment:
This should read:

```
validates :bar, presence: true
validates :user, presence: true
```

Get rid of the `_id`s there! This will cause ActiveRecord to actually do the extra work of making sure there's a matching bar/user in the database - instead of just checking to make sure that there's an integer in this column.